comp/core/sysprobeconfig/component.go

```diff
@@ -0,0 +1,132 @@
+
+	// GetString gets a string-typed config parameter value.
+	GetString(key string) string
+
+	// GetBool gets an integer-typed config parameter value.
+	GetBool(key string) bool
+
+	// GetInt gets an integer-typed config parameter value.
+	GetInt(key string) int
+
+	// GetInt32 gets an int32 config parameter value.
+	GetInt32(key string) int32
+
+	// GetInt64 gets an int64 config parameter value.
+	GetInt64(key string) int64
+
+	// GetFloat64 gets an float64 config parameter value.
+	GetFloat64(key string) float64
+
+	// GetTime gets a time as a config parameter value.
+	GetTime(key string) time.Time
+
+	// GetDuration gets a duration as a config parameter value, parsing common suffixes.
+	// Bare integers are treated as nanoseconds.
+	GetDuration(key string) time.Duration
+
+	// GetStringSlice gets a slice of strings (represented as a list in the source YAML)
+	GetStringSlice(key string) []string
+
+	// GetFloat64SliceE gets a slice of floats (represented as a list in the source
+	// YAML) as a config parameter value, returning an error if any of the
+	// values cannot be parsed
+	GetFloat64SliceE(key string) ([]float64, error)
+
+	// GetStringMap gets a map (represented as an object in the source YAML) as
+	// a config parameter value.
+	GetStringMap(key string) map[string]interface{}
+
+	// GetStringMapString gets a map (represented as an object in the source
+	// YAML) as a config parameter value, treating each value as a string.
+	GetStringMapString(key string) map[string]string
+
+	// GetStringMapStringSlice gets a map (represented as an object in the source
+	// YAML) as a config parameter value, treating each value as a string slice
+	// (represented as an array in the source YAML).
+	GetStringMapStringSlice(key string) map[string][]string
+
+	// GetSizeInBytes gets a size as a config parameter value, parsing common suffixes.
+	GetSizeInBytes(key string) uint
+
+	// AllSettings merges all settings and returns them as a map[string]interface{}.
+	AllSettings() map[string]interface{}
+
+	// AllSettingsWithoutDefault is like AllSettings but omits configuration that was
+	// applied based on defaults.
+	AllSettingsWithoutDefault() map[string]interface{}
+
+	// AllKeys returns all keys holding a value, regardless of where they are
+	// set. Nested keys are returned with a v.keyDelim separator
+	AllKeys() []string
+
+	// GetKnownKeys returns all the keys that meet at least one of these criteria:
+	// 1) have a default, 2) have an environment variable binded, 3) are an alias or 4) have been SetKnown()
+	GetKnownKeys() map[string]interface{}
+
+	// GetEnvVars returns a list of the env vars that the config supports.
+	// These have had the EnvPrefix applied, as well as the EnvKeyReplacer.
+	GetEnvVars() []string
+
+	// IsSectionSet checks if a given section is set by checking if any of
+	// its sub-keys is set.
+	IsSectionSet(section string) bool
+
+	// Warnings returns config warnings collected during setup.
+	Warnings() *config.Warnings
+}
```

Comment:
This interface seems to be the same as `comp/core/config/component.go`. You can use  `type Component config.Component` if they must be the same or the following code if this config must have all the methods from config.Component and other ones. 
``
type Component interface {
	config.Component
}
``
Note: See my comment comp/core/sysprobeconfig/config.go before applying a change.